openpype/modules/deadline/plugins/publish/submit_houdini_cache_deadline.py

```diff
@@ -0,0 +1,181 @@
+
+    Submission is done through the Deadline Web Service as
+    supplied via the environment variable AVALON_DEADLINE.
+
+    """
+
+    label = "Submit Scene to Deadline"
+    order = pyblish.api.IntegratorOrder
+    hosts = ["houdini"]
+    families = ["publish.hou"]
+    targets = ["local"]
+
+    priority = 50
+    jobInfo = {}
+    pluginInfo = {}
+    group = None
+
+    def get_job_info(self):
+        job_info = DeadlineJobInfo(Plugin="Houdini")
+
+        job_info.update(self.jobInfo)
+        instance = self._instance
+        context = instance.context
+        assert all(
+            result["success"] for result in context.data["results"]
+        ), "Errors found, aborting integration.."
+
+        # Deadline connection
+        AVALON_DEADLINE = legacy_io.Session.get(
+            "AVALON_DEADLINE", "http://localhost:8082"
+        )
+        assert AVALON_DEADLINE, "Requires AVALON_DEADLINE"
+
+        project_name = instance.context.data["projectName"]
+        filepath = context.data["currentFile"]
+        scenename = os.path.basename(filepath)
+        job_name = "{scene} - {instance} [PUBLISH]".format(
+            scene=scenename, instance=instance.name)
+        batch_name = "{code} - {scene}".format(code=project_name,
+                                               scene=scenename)
+        if is_in_tests():
+            batch_name += datetime.now().strftime("%d%m%Y%H%M%S")
+
+        job_info.Name = job_name
+        job_info.BatchName = batch_name
+        job_info.Plugin = instance.data["plugin"]
+        job_info.UserName = context.data.get("deadlineUser", getpass.getuser())
+        rop_node = self.get_rop_node(instance)
+        if rop_node.type().name() != "alembic":
+            frames = "{start}-{end}x{step}".format(
+                start=int(instance.data["frameStart"]),
+                end=int(instance.data["frameEnd"]),
+                step=int(instance.data["byFrameStep"]),
+            )
+
+            job_info.Frames = frames
+
+        job_info.Pool = instance.data.get("primaryPool")
+        job_info.SecondaryPool = instance.data.get("secondaryPool")
+
+        attr_values = self.get_attr_values_from_data(instance.data)
+
+        job_info.ChunkSize = instance.data["chunkSize"]
+        job_info.Comment = context.data.get("comment")
+        job_info.Priority = attr_values.get("priority", self.priority)
+        job_info.Group = attr_values.get("group", self.group)
+
+        keys = [
+            "FTRACK_API_USER",
+            "FTRACK_API_KEY",
+            "FTRACK_SERVER"
+        ]
+
+        # Add OpenPype version if we are running from build.
+        if is_running_from_build():
+            keys.append("OPENPYPE_VERSION")
+        # Add mongo url if it's enabled
+        if self._instance.context.data.get("deadlinePassMongoUrl"):
+            keys.append("OPENPYPE_MONGO")
+
+        environment = dict({key: os.environ[key] for key in keys
```

Comment:
Could we leverage `acre` here? Also, I think there's a flag for Deadline which will pass the current environment to the job (or I might be misremembering)

Reply:
> a flag for Deadline which will pass the current environment to the job

do you mean `job_info.add_render_job_env_var()` ?

Reply:
That seems like single env vars, maybe I remember wrong, but when you submit to deadline, you have the option to pass the whole of the host environment as is

Reply:
We don't want to pass the full environment because we do still want the deadline machine to 'build their OS-compatible environment' for the application, etc. - There are just specific vars that we DO want to pass along in certain situations. That's why we need to be explicit about it.